### PR TITLE
Fix skipping tests in pipelines Facebook and Webex functional tests

### PR DESCRIPTION
**Pull request by BruceHaley**
This fixes the broken pipeline Botbuilder-DotNet-Facebook-Functional-Tests-yaml, making the test (1) execute and (2) pass.

This also fixes the broken pipeline BotBuilder-DotNet-Webex-Functional-Test-yaml, making the test (1) execute and (2) pass.

Re execution: The test modules have been calling "Assert.Inconclusive()". That causes the test to be skipped. Thus, the build passes when it should be failing.

### CODE REVIEW

**Comment by mrivera-ms** [comment]
@Virtual-Josh , please take a look.

File: FunctionalTests/Microsoft.Bot.Builder.FunctionalTests/FacebookChatTests.cs [hunk 1]
```diff
@@ -132,25 +132,25 @@ private void GetEnvironmentVars()
                 _appSecret = Environment.GetEnvironmentVariable("FacebookTestBotFaceBookAppSecret");
                 if (string.IsNullOrWhiteSpace(_appSecret))
                 {
-                    Assert.Inconclusive("Environment variable 'FacebookTestBotFaceBookAppSecret' not found.");
+                    Assert.Fail("Environment variable 'FacebookTestBotFaceBookAppSecret' not found.");
```
Comment (Virtual-Josh) [inline FunctionalTests/Microsoft.Bot.Builder.FunctionalTests/FacebookChatTests.cs:135]: For all the `inconclusive` to `fail` changes, is there a situation where some of these env vars could not be set and that's the expected behavior? This change could possibly create a false negative.

**Comment by mrivera-ms** [comment]
Once the tests are not skipped, they still fail. Created Issue #4545 to track a fix.

File: libraries/Adapters/Microsoft.Bot.Builder.Adapters.Facebook/FacebookClientWrapper.cs [hunk 2]
```diff
@@ -86,8 +87,13 @@ public virtual async Task<string> SendMessageAsync(string path, FacebookMessage
                         var stringResponse = JsonConvert.DeserializeObject<FacebookResponseOk>(responseBody);
                         return stringResponse.MessageId;
                     }
+                    else
+                    {
+                        // Return error info
+                        return res.ToString();
+                    }
```
Comment (mrivera-ms) [inline libraries/Adapters/Microsoft.Bot.Builder.Adapters.Facebook/FacebookClientWrapper.cs:94]: This 'else' is not needed. 

Also, it seems like the method's intent is to return either a message Id, if the request is successful, or an empty string, if not. This introduces a change of behavior. Is that what you mean?

File: libraries/Adapters/Microsoft.Bot.Builder.Adapters.Facebook/FacebookClientWrapper.cs [hunk 3]
```diff
@@ -11,6 +11,7 @@
 using Microsoft.AspNetCore.Http;
 using Microsoft.Bot.Builder.Adapters.Facebook.FacebookEvents;
 using Microsoft.Bot.Builder.Adapters.Facebook.FacebookEvents.Handover;
+using Microsoft.Bot.Schema;
```
Comment (mrivera-ms) [inline libraries/Adapters/Microsoft.Bot.Builder.Adapters.Facebook/FacebookClientWrapper.cs:14]: What is this added for?

File: FunctionalTests/Microsoft.Bot.Builder.FunctionalTests/FacebookChatTests.cs [hunk 4]
```diff
@@ -48,7 +48,7 @@ private async Task SendMessageAsync(string echoGuid)
                 request.Method = HttpMethod.Post;
                 request.RequestUri = new Uri(_botEndpoint);
 
-                await client.SendAsync(request);
+                var response = await client.SendAsync(request);
```
Comment (mrivera-ms) [inline FunctionalTests/Microsoft.Bot.Builder.FunctionalTests/FacebookChatTests.cs:51]: Is 'response' used anywhere?

File: FunctionalTests/Microsoft.Bot.Builder.FunctionalTests/WebexClientTest.cs [hunk 5]
```diff
@@ -100,41 +100,47 @@ private async Task RefreshAccessToken()
...
-            _userAccessToken = kvPairs["access_token"];
+            try
+            {
+                var kvPairs = JsonConvert.DeserializeObject<Dictionary<string, string>>(response);
+                _userAccessToken = kvPairs["access_token"];
+            }
+            catch
+            {
+                throw new Exception($"RefreshAccessToken() failed: response = {response}");
+            }
```
Comment (mrivera-ms) [inline FunctionalTests/Microsoft.Bot.Builder.FunctionalTests/WebexClientTest.cs:111]: Why throw a new exception? If the operation throws, the test will fail with the error needed. If you want to wrap the exception, you can add the original exception as an inner exception so the information is not lost.

**Review by mrivera-ms — DISMISSED** [review]

File: FunctionalTests/Microsoft.Bot.Builder.FunctionalTests/FacebookChatTests.cs (hunk 1 above)
Comment (BruceHaley) [reply FunctionalTests/Microsoft.Bot.Builder.FunctionalTests/FacebookChatTests.cs:135]: No.

File: libraries/Adapters/Microsoft.Bot.Builder.Adapters.Facebook/FacebookClientWrapper.cs (hunk 2 above)
Comment (BruceHaley) [reply libraries/Adapters/Microsoft.Bot.Builder.Adapters.Facebook/FacebookClientWrapper.cs]: Sorry, that was a way to debug. I have now changed this to throw an exception so the error is visible in Application Insights

File: FunctionalTests/Microsoft.Bot.Builder.FunctionalTests/FacebookChatTests.cs (hunk 4 above)
Comment (BruceHaley) [reply FunctionalTests/Microsoft.Bot.Builder.FunctionalTests/FacebookChatTests.cs]: Good point. That was another debug effort. Now removed.

File: FunctionalTests/Microsoft.Bot.Builder.FunctionalTests/WebexClientTest.cs (hunk 5 above)
Comment (BruceHaley) [reply FunctionalTests/Microsoft.Bot.Builder.FunctionalTests/WebexClientTest.cs]: Thanks for the suggestion. Inner exception now captured.

File: libraries/Adapters/Microsoft.Bot.Builder.Adapters.Facebook/FacebookClientWrapper.cs (hunk 3 above)
Comment (BruceHaley) [reply libraries/Adapters/Microsoft.Bot.Builder.Adapters.Facebook/FacebookClientWrapper.cs:14]: Good catch. Deleted.

**Comment by BruceHaley** [comment]
> Once the tests are not skipped, they still fail. Created Issue #4545 to track a fix.

Facebook tests passing. Webex still failing.

**Comment by BruceHaley** [comment]
Webex tests were likely fixed when Webex webhook was re-created.

**Comment by BruceHaley** [comment]
Closed in favor of duplicate PR #4583

**Pull request closed without merging**